### DISCUSSION TITLE
Please could these 3 Unknown Flashroms be added to v6.2 if possible?

### DISCUSSION THREAD

**Discussion by fiatbravodriver** [sanni/cartreader#155] [Need help]
![n64 unknown flashroms](https://user-images.githubusercontent.com/43493640/119119976-a7657d00-ba23-11eb-9a40-f2fb09e46c7c.jpg)

Hi please would it be possible for these 3 flashroms flagged as unknown to be added to the next release for N64 Repos. 

I am more than happy to test if an Updated n64.ino could be supplied? I am not very good at changing code.

Many thanks

Update:

**Here is a better Photo of the First Example:**

![image](https://user-images.githubusercontent.com/43493640/119258171-80907d80-bbc0-11eb-8835-ec20a40a2a53.png)

Here is a Link to the Data Sheet:

https://static6.arrow.com/aropdfconversion/7c4e916c14c9ea45d389399aeec086fb834f339b/_s29gl032n_64_mbit_32_mbit_3_v_page_mode_mirrorbit_flash.pdf

**I have found the datasheet for the 3rd Example:**

https://datasheetspdf.com/pdf/1303569/Micron/M29W128GL/1

![image](https://user-images.githubusercontent.com/43493640/119257905-4d012380-bbbf-11eb-92b2-14321491ad38.png)

**Comment by sanni** [comment]
I can't add any repros that I don't own but I did add a new manual config menu that allows you to try the existing flash functions with unknown flashroms. This will only work if the unknown flashrom is the same kind as the supported ones just with a different size/id.

227E 1000: Try 8MB flash size with 32 byte buffer and 8kbyte sector size
1240 1240: this ID usually means that the cartridge was not inserted perfectly centered
227E 2100: Try 16MB flash size with 64 byte buffer and 128 kbyte sector size

**Comment by fiatbravodriver** [comment]
Fantastic feature to add thank you. 

Unfortunately 227E 1000: Try 8MB size with 32 byte buffer - Gives Erasing...failed error
1240 1240 - You were perfectly correct All solved Thank you. 
227E 2100 - Worked Perfectly

If there is anything I can try for 227E 1000 please let me know. 

Thanks for taking the time with all your help and this new feature.

**Comment by sanni** [comment]
I have added a new menu for selecting the sector size. Try 8KB sector size with the 227E 1000.

**Reply by fiatbravodriver** [reply]
Thank you. I updated to 6.3 and copied all the SD files. I selected : 8mb size , 32 byte buffer with 8kb sector size. The device is just displaying: File size 8mb Erasing... 

It has been doing this for about 25mins. I suspect there is a problem. 

I think I may have to power off.

**Comment by sanni** [comment]
I think I fixed the erase problem now. It should only take like 5 minutes and blink alot.

**Reply by fiatbravodriver** [reply]
Thanks, I am getting a "5 bytes did not verify . Press button" message. I will try a few more times.

**Comment by sanni** [comment]
You can also try 64KB sector size to see if that makes any difference.

**Reply by fiatbravodriver** [reply]
thanks for the option. I am getting "7 bytes did not verify". Happy to keep trying.

**Reply by fiatbravodriver** [reply]
Tried again on a mains power and no verify error :-)

**Comment by sanni** [comment]
What sector size did you end up using? I'll add it to the known flashrom list once you confirmed a working config.

**Reply by fiatbravodriver** [reply]
I forgot :-( so I did another flash 8mb size , 32 byte buffer with 8kb sector size and it Worked :-)
Thanks again.
